templates/install_puppet.bat.erb

```diff
@@ -0,0 +1,11 @@
+:wait_for_pid
+FOR /F "tokens=*" %%A IN ('tasklist /FI "PID eq <%= @puppet_agent_pid %>"  /NH') DO set _task=%%A
+echo %_task% | findstr "No tasks are running" >nul
+IF NOT %errorlevel% == 0 ( GOTO wait_for_pid )
+
+
+msiexec.exe /qn /i <%= @_msi_location %> /l*v <%= @_logfile %>
```

Comment:
@cyberious you always should use `/norestart`. There is almost never a good reason not to. It doesn't hurt anything if it is there, but it will prevent Windows from moving forward with a previous pending reboot